Dockerfile

```diff
@@ -6,9 +6,16 @@ RUN apt-key adv --keyserver keyserver.ubuntu.com --recv E56151BF && \
     apt-get update && \
     apt-get install --no-install-recommends -y --force-yes mesos=0.23.0-1.0.debian81 sbt
 
-COPY . /marathon
 WORKDIR /marathon
 
+# The build configuration including dependencies changes
+# less frequently than the source code. By separating
+# these steps we can greatly speed up cached local docker builds
+ADD project /marathon/project
```

Comment:
Are you using `ADD` instead of `COPY` for any particular reason?

According to Docker, `COPY` should be preferred over `ADD` (https://docs.docker.com/articles/dockerfile_best-practices/#add-or-copy):

> Although ADD and COPY are functionally similar, generally speaking, COPY is preferred. That’s because it’s more transparent than ADD. COPY only supports the basic copying of local files into the container, while ADD has some features (like local-only tar extraction and remote URL support) that are not immediately obvious. Consequently, the best use for ADD is local tar file auto-extraction into the image, as in ADD rootfs.tar.xz /.